File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int> curLen;
        int res = 0;

        for (int num : nums) {
            if (curLen[num]) {
                continue;
            }

            curLen[num] = curLen[num-1] + curLen[num+1] + 1;
            curLen[num - curLen[num-1]] = curLen[num];
            curLen[num + curLen[num+1]] = curLen[num];

            // cout << num << " " << curLen[num] << endl;

            res = max(res, curLen[num]);   
        }

        return res;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int res = 0;

        for (int num : nums) {
            set.add(num);
        }

        for (int num : set) {
            if (set.contains(num-1)) {
                continue;
            }

            int combo = 1;

            while (set.contains(num+1)) {
                combo ++;
                num = num+1;
            }

            res = Math.max(res, combo);
        }

        return res;
    }
}